FEATURE_PROCESS.adoc

```diff
@@ -1,5 +1,106 @@
+
+This process should be followed once a developer anticipates beginning work on a feature in the next three months.
+
+NOTE: Requesting new features or recording ideas for new features are *not* part of this process. Feature requests are handled by simply filing an issue in the relevant project JIRA, typically https://issues.redhat.com/projects/WFLY/summary[WFLY].
+
+== The Process
+
+Feature development proceeds in five stages:
+
+=== Kickoff
+
+When a developer intends to begin work on a feature they should do the following:
+
+* If one doesn't already exist, file an issue in the relevant project JIRA, typically https://issues.redhat.com/projects/WFLY/summary[WFLY], https://issues.redhat.com/projects/WFCORE/summary[WFCORE] or https://issues.redhat.com/projects/HAL/summary[HAL].
+* File an issue with the https://github.com/orgs/wildfly/projects/7[WildFly Feature Planning project]. To do this create an issue using the https://github.com/wildfly/wildfly-proposals/issues/new?assignees=&labels=feature&projects=wildfly%2F7&template=feature-development.yaml[Feature Development issue template].
+* Draft a minimal feature analysis, based on the current https://github.com/wildfly/wildfly-proposals/blob/main/design-doc-template.adoc[design-doc-template]. This should include the expected stability level, links to the issues above, and enough Overview and User Story information to give people interested in being part of the feature team a sense of what would be involved.
+* Submit a draft analysis PR to the https://github.com/wildfly/wildfly-proposals[`wildfly-proposals`] repository.
+* Send an email to the `wildfly-dev` mail list telling people about the upcoming work and asking for volunteers to participate in the feature team.
+
+When the above tasks are complete, change the `Status` field of the WildFly Feature Planning project JIRA to `Planning`.
+
+=== Planning
+
+The Planning stage of a feature consists of <<feature-team,feature team>> formation and enough discussion of the feature among the team that they can agree on a target feature stability level and target WildFly release in which they expect to complete the feature.
+
+NOTE: The target WildFly release is just a target and is not a requirement.
+
+Update the feature analysis document with the planned stability level (if it has changed) and the GitHub ids of the feature team members.
+
+Set the `Iteration` field to the iteration for the WildFly release.
+
+When the above tasks are complete, change the `Status` field of the WildFly Feature Planning project JIRA to `In Progress`.
+
+IMPORTANT: Changes to the `Iteration` field of a WildFly Feature Planning project issue should always reflect the consensus of the feature team.
```

Comment:
Is it also worth some comment that we only plan to have iterations for releases N and N+1 so if a feature is a longer running activity they may not be able to set the iteration?

Also do we consider iteration an "exit criteria" for this stage or useful info?

Reply:
Done.